src/Course/List.hs

```diff
@@ -239,11 +226,14 @@ flattenAgain =
 --
 -- >>> seqOptional (Empty :. map Full infinity)
 -- Empty
-seqOptional ::
-  List (Optional a)
-  -> Optional (List a)
-seqOptional =
-  error "todo: Course.List#seqOptional"
+seqOptional :: forall a . Show a => List (Optional a) -> Optional (List a)
```

Comment:
the typeclass constraint `Show a =>` that we used for debugging isn't needed now (its function `show` isn't used here).
This is why its better to use [Debug.Trace](https://hackage.haskell.org/package/base-4.12.0.0/docs/Debug-Trace.html) module, as it leaves warnings where its used.